packages/mask/src/extension/popups/pages/Wallet/WalletGuard/usePaymentPasswordGuard.ts

```diff
@@ -0,0 +1,15 @@
+import { PopupRoutes } from '@masknet/shared-base'
+import { useEffect } from 'react'
+import { useMatch, useNavigate } from 'react-router-dom'
+import { useHasPassword } from '../../../hooks/index.js'
+
+export function usePaymentPasswordGuard() {
+    const { hasPassword, loading } = useHasPassword()
+    const matchSetPaymentPassword = !!useMatch(PopupRoutes.SetPaymentPassword)
+    const navigate = useNavigate()
+
+    useEffect(() => {
+        if (matchSetPaymentPassword || hasPassword || loading) return
+        navigate(PopupRoutes.SetPaymentPassword)
```

Comment:
Use callback or pre-calculated route.

Reply:
Like the original implement?

Then attacker can bypass the lock by a visiting certain route to access to user's data.